test/chat_api_web/controllers/note_controller_test.exs

```diff
@@ -0,0 +1,123 @@
+defmodule ChatApiWeb.NoteControllerTest do
+  use ChatApiWeb.ConnCase
+
+  import ChatApi.Factory
+  alias ChatApi.Notes.Note
+
+  @update_attrs %{
+    body: "updated body!"
+  }
+  @invalid_attrs %{body: nil, customer_id: nil}
+
+  setup %{conn: conn} do
+    account = insert(:account)
+    agent = insert(:user, account: account)
+    customer = insert(:customer, account: account)
+    note = insert(:note, body: "note body", account: account, author: agent, customer: customer)
+
+    conn = put_req_header(conn, "accept", "application/json")
+    authed_conn = Pow.Plug.assign_current_user(conn, agent, [])
+
+    {:ok,
+     conn: conn,
+     authed_conn: authed_conn,
+     account: account,
+     agent: agent,
+     customer: customer,
+     note: note}
+  end
+
+  describe "index" do
+    test "lists all notes for the specified customer in the current account", %{
+      authed_conn: authed_conn,
+      note: note
+    } do
+      conn =
+        get(authed_conn, Routes.note_path(authed_conn, :index), %{
+          "note" => %{customer_id: note.customer_id}
+        })
+
+      note_ids = json_response(conn, 200)["data"] |> Enum.map(& &1["id"])
+
+      assert note_ids == [note.id]
+    end
```

Comment:
can we also add a test to see what happens when a `customer_id` param is missing from the request?

(ideally, it would return a `400` error)